File: Programmers/로또의 최고 순위와 최저 순위.js
function solution(lottos, win_nums) {
    //내가 고른 숫자를 오름차순으로 정렬
    const mySortNum  = lottos.sort((a,b)=>a-b);
    //배열에서 0을 제외함
    const expZero = [];
    for(let i=0;i<mySortNum.length;i++){
        if(mySortNum[i] === 0){
            ''
        }else{
            expZero.push(mySortNum[i])
        }
    }
    //배열에서 0의 개수
    const zeroNum = mySortNum.length-expZero.length
    //당첨 숫자를 오름차순으로 정렬
    const winSortNum = win_nums.sort((a,b)=>a-b);
    //당첨 숫자와 일치하는 개수를 찾아 최소 등수를 구함
    let winMin = 7;
    for(let k=0;k<expZero.length;k++){
        if(expZero.length === 0){
            ''
        }else{
            for(let p=0;p<winSortNum.length;p++){
                if(expZero[k] === winSortNum[p]){
                   winMin -= 1;
                }
            }
    }}
    //당첨된 최소 개수에 0의 개수를 빼서 최대 등수를 구함.
    //1개와 2개 당첨 모두 6등으로 처리되게 함.
    let winMax = winMin - zeroNum;
      if(winMax === 7){
        winMax = 6
    }
        if(winMin === 7){
        winMin = 6
    }
    return([winMax,winMin])
 
}

File: Py/로또의 최고 순위와 최저 순위/로또의 최고 순위와 최저 순위.py
def solution(lottos, win_nums):
  #lottos에서의 0의 갯수, 0이 아닌 숫자를 따로 분리
    zero_num = 0
    lottosNotZero = []
    for i in range(0,len(lottos)):
        if(lottos[i] == 0):
            zero_num += 1
            #win_nums에 있다면 lottosNotZero에 추가
        elif lottos[i] in win_nums:
            lottosNotZero.append(lottos[i])
    #가장 순위가 낮을 경우 = 7-lottosNotZero에 있는 수
    minRank = 7 - len(lottosNotZero)
    #가장 순위가 높을 경우 = 0이 다 맞았을 때
    maxRank = minRank - zero_num
    #모두 맞추지 않았을 때 7로 나오는 것을 방지
    if minRank == 7:
        minRank = 6
    if maxRank == 7:
        maxRank = 6
    answer = [maxRank,minRank]
    return answer
